src/zenml/client.py

```diff
@@ -364,6 +364,30 @@ def _reset_instance(cls, client: Optional["Client"] = None) -> None:
         """
         cls._global_client = client
 
+    @staticmethod
+    def _fail_for_sql_zen_store(method):
+        """Decorator for all methods, that are disallowed when the client is not connected through REST API.
+
+        Args:
+            method: The method
+
+        Returns:
+            The decorated method.
+        """
+
+        @functools.wraps(method)
+        def wrapper(self, *args, **kwargs):
+            # No isinstance check to avoid importing ZenStore implementations
+            breakpoint()
+            if self.zen_store.__class__.__name__ == "SqlZenStore":
+                raise TypeError(
+                    "This method is not allowed when not connected "
+                    "to a ZenML Server through the API interface.."
+                )
+            return method(*args, **kwargs)
```

Comment:
```suggestion
        @functools.wraps(method)
        def wrapper(self, *args, **kwargs):
            # No isinstance check to avoid importing ZenStore implementations
            if self.zen_store.__class__.__name__ == "SqlZenStore":
                raise TypeError(
                    "This method is not allowed when not connected "
                    "to a ZenML Server through the API interface.."
                )
            return method(self, *args, **kwargs)
```